src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -2514,22 +2514,42 @@ internal override Symbol RemapSymbolIfNecessaryCore(Symbol symbol)
 
         internal override bool ShouldSkipSyntaxNodeAnalysis(SyntaxNode node, ISymbol containingSymbol)
         {
-            if (containingSymbol.Kind is SymbolKind.Method)
+            switch (containingSymbol.Kind)
             {
-                switch (node)
-                {
-                    case TypeDeclarationSyntax:
-                        // Skip the topmost type declaration syntax node when analyzing primary constructor
-                        // to avoid duplicate syntax node callbacks.
-                        // We will analyze this node when analyzing the type declaration type symbol.
-                        return true;
+                case SymbolKind.Method:
+                    switch (node)
+                    {
+                        case TypeDeclarationSyntax:
+                            // Skip the topmost type declaration syntax node when analyzing primary constructor
+                            // to avoid duplicate syntax node callbacks.
+                            // We will analyze this node when analyzing the type declaration type symbol.
+                            return true;
+
+                        case CompilationUnitSyntax:
+                            // Skip compilation unit syntax node when analyzing synthesized top level entry point method
+                            // to avoid duplicate syntax node callbacks.
+                            // We will analyze this node when analyzing the global namespace symbol.
+                            return true;
+
+                        case BaseListSyntax:
+                            // Skip the base list syntax node when analyzing primary constructor
+                            // to avoid duplicate syntax node callbacks.
+                            // We will analyze this node when analyzing the type declaration type symbol.
+                            return true;
+                    }
+
+                    break;
 
-                    case CompilationUnitSyntax:
-                        // Skip compilation unit syntax node when analyzing synthesized top level entry point method
+                case SymbolKind.NamedType:
+                    if (node is PrimaryConstructorBaseTypeSyntax)
```

Comment:
>if (node is PrimaryConstructorBaseTypeSyntax) [](http://example.com/codeflow?start=20&length=45)

Is it possible to subscribe to an event for one of the constituents of this node? For example, for the `ArgumentList` or one of the arguments. How many call backs are we going to get then?

Reply:
What about other parts of the type declaration header?

Reply:
> Is it possible to subscribe to an event for one of the constituents of this node? For example, for the ArgumentList or one of the arguments. How many call backs are we going to get then?

Thanks, I enhanced the added unit test to verify we get single callbacks for all the descendants of BaseListSyntax. There is existing code which ensures we only analyze ArgumentList and its descendants when processing the primary constructor method symbol: https://github.com/dotnet/roslyn/blob/c5d20779f421caafbd4e18e3bfc5c0320a0a29f8/src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs#L2432-L2505